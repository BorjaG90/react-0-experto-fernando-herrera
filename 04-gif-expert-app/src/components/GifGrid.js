import React from 'react'

// import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

  // const [images, setImages] = useState([]);
  const {loading} = useFetchGifs();

  console.log(loading);

  // useEffect( () => {
  //   getGifs(category)
  //     .then(images => setImages(images)); 
  // }, [ category ]);

  return (
    <>
      <h3>{category}</h3>

      {loading ? 'Cargando...': 'Data cargada'}

      {/* <div className='card-grid'>
        {
          images.map((image) => (
            <GifGridItem
            key={image.id}
            { ...image }
            />
            ))
          }
      </div> */}
    </>
  )
}
