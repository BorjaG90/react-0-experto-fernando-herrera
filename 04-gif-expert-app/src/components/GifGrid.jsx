export const GifGrid = ({category, key}) => {
  const gifs = [1,2,3,4,5]

  return (
    <>
      <h3>{category}</h3>
      {
        gifs.map(gif => {
          <p></p>
        })
      }
    </>
  )
}
