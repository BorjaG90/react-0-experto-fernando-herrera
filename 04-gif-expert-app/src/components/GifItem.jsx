export const GifItem = ({image, url, title}) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}
