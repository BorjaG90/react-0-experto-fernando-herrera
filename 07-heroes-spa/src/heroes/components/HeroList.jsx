import { getheroesByPublisher } from "../helpers"

export const HeroList = ({publisher}) => {
  const heroes = getheroesByPublisher(publisher);

  return (
    <ul>
      {heroes.map(hero => (
        <li>{hero.superhero}</li>
      ))}
    </ul>
  )
}
