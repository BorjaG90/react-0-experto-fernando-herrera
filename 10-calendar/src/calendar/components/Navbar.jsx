export const Navbar = () => {
  return (
    <div className="navbar navbar-dark bg-dark mb-4 px-4">
      <span className="">
        <i className="fas fa-calendar-alt"></i>
        &nbsp;
        Borja
      </span>
      <button className="btn btn-outline-danger">
        <i className="fas fa-sign-out-alt"></i>
        <span>Salir</span>
      </button>
    </div>
  )
}
