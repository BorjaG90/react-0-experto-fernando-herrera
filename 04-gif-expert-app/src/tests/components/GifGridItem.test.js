import {shallow} from 'enzyme'
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => { 
  const title = 'Un tÃ­tulo';
  const url = 'http://localhost/algo.jpg';

  test('should de mostrar el componente correctamente', () => { 
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect ( wrapper ).toMatchSnapshot();
   })
 })