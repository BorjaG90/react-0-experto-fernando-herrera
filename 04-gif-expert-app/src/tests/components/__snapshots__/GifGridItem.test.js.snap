// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un título"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
